src/routes/v2/address.ts

```diff
@@ -265,9 +265,22 @@ async function detailsSingle(
     }
 
     // Query the Insight API.
-    const retData = await detailsFromInsight(address, currentPage)
+    let retData: any = await detailsFromInsight(address, currentPage)
+    retData = retData.data
 
-    // Return the array of retrieved address information.
+    // Calculate pagesTotal from response
+    const pagesTotal = Math.ceil(retData.txApperances / PAGE_SIZE)
+
+    // Append different address formats to the return data.
+    retData.legacyAddress = BITBOX.Address.toLegacyAddress(retData.addrStr)
+    retData.cashAddress = BITBOX.Address.toCashAddress(retData.addrStr)
+    delete retData.addrStr
+
+    // Append pagination information to the return data.
+    retData.currentPage = currentPage
+    retData.pagesTotal = pagesTotal
+
+    // Return the retrieved address information.
```

Comment:
I had to pull this data processing out because this function needs to return a Promise instead of a resolved value, in order to be parallelized.